src/transformers/models/mobilebert/modeling_tf_mobilebert.py

```diff
@@ -1026,6 +1030,33 @@ def __init__(self, config, *inputs, **kwargs):
     def get_output_embeddings(self):
         return self.mobilebert.embeddings
 
+    def resize_token_embeddings(self, new_num_tokens):
+        super().resize_token_embeddings(new_num_tokens=new_num_tokens)
+
+        if new_num_tokens is not None:
+            num_tokens_to_copy = min(self.predictions.predictions.bias.shape[0], new_num_tokens)
+            self.predictions.predictions.vocab_size = num_tokens_to_copy
+            init_bias = tf.zeros((new_num_tokens,))
+            init_bias[:num_tokens_to_copy] = self.predictions.predictions.bias.value()[:num_tokens_to_copy]
+            name = self.name + "/" + self.predictions.name + "/" + self.predictions.predictions.name + "/bias"
+            self.predictions.predictions.bias = self.add_weight(
+                shape=(new_num_tokens,), initializer="zeros", trainable=True, name=name
+            )
+            self.predictions.predictions.bias.assign(init_bias)
+
+            init_weights = tf.zeros((new_num_tokens,))
+            init_weights[:num_tokens_to_copy] = self.predictions.predictions.decoder.value()[:num_tokens_to_copy]
+            name = (
+                self.name + "/" + self.predictions.name + "/" + self.predictions.predictions.name + "/decoder/weight"
+            )
+            self.predictions.predictions.decoder = self.add_weight(
+                shape=(new_num_tokens, self.predictions.predictions.config.embedding_size),
+                initializer="zeros",
+                trainable=True,
+                name=name,
+            )
+            self.predictions.predictions.decoder.assign(init_weights)
```

Comment:
I was hoping to avoid a special case for MobileBERT by resizing the result of `get_output_embeddings()` in the generic function when it's different from `get_input_embeddings()`.
Otherwise we will also need a special function for T5 and mT5 when `tied_word_embeddings=False`.

Reply:
So for the bias+decoder weights, to add a test in the parent `resize_token_embeddings` like:
```
if self.get_input_embeddings() != self.get_outpu_embeddings():
    #### doing the resize of the decoder weights + bias
else:
   ### doing the resize of the bias
```

Is that what you meant?

Reply:
Not exactly, the bias should be kept in its own part like you did. Just add a 
```
if self.get_input_embeddings() != self.get_output_embeddings():
    #### doing the resize of the decoder weights
```
should be fine.

Reply:
Perfect! Working on this!!

Reply:
Ok what do you think now?